merkledag/merkledag.go

```diff
@@ -93,6 +103,23 @@ func (n *dagService) Get(ctx context.Context, c *cid.Cid) (*Node, error) {
 	return res, nil
 }
 
+func (n *dagService) GetLinks(ctx context.Context, c *cid.Cid) ([]*Link, error) {
+	node, err := n.Get(ctx, c)
+	if err != nil {
+		return nil, err
+	}
+	return node.Links, nil
+}
+
+func (n *dagService) GetOfflineLinkService() LinkService {
```

Comment:
I did it this way to enforce type safety.  Otherwise we could be passed a LinkService and get a DAGService.  If you really don't like that I can change it.